### ISSUE TITLE
Prevent string matching only in Open Refine when adding to Wikidata.

### ISSUE THREAD

**Issue by andrawaag** [OpenRefine/OpenRefine#4552]
I am often faced with OpenRefine edits in Wikidata which were clearly caused by string matching only. I know that Open Refine suggests possible semantic types, but it is rather easy to ignore those facets and move forward by disambiguation on strings only. Here is an example: https://www.wikidata.org/w/index.php?title=Q29733890&type=revision&diff=1246714572&oldid=1237282345 When using OR 3.4 an item on a gene was enriched with statements on a Spanish cultural heritage object.  If this mismatch is not caught early, it can be quite challenging to rollback, since subsequent edits will be based on these edits and it is not always clear if the heuristics used are based on the original semantic type, or the newly introduced one. 

### Proposed solution
Writing to wikidata should only be possible after disambiguation by statements only. The labels and descriptions from Wikidata should be treated as notes, but not as input for disambiguation pipelines. Being able to resolve on strings only can be very powerful in cleaning data, but again not in pipelines to enrich Wikidata (or other Knowledge graphs). It would be really helpful if OR prevents such uploads

**Comment by egonw** [comment]
Yes, very much seconded. Matching data based on String matches only works if you are 100% it is about exactly the same thing, and that rarely is the same (not even for drugs). Over the years I have done quite some data curation because of linking data based on string matches. If a string match is the only reason to "refine", I recommend a strong warning: "OpenRefine is not responsible for inaccurate matching; all incorrect matches may have to be manually cleaned". The solution that Andra suggests is the inverse of that: only manually curated matches go into Wikidata.

**Comment by thadguidry** [comment]
I think there might be several reasons as to why this happened.

I don't think we can quickly jump to conclusions about string matching only for that upload.  I myself would love to know more details about the reconcile itself that the user did and tied to the edit but that isn't part of the batch uploads to the service currently (for instance poking around or looking at the [catalog upload](https://editgroups.toolforge.org/b/OR/e3940410f4a/).

For instance, we might consider that **autoMatch** might play a big part (or small part) in this user error and it is something that we don't provide with the uploaded edits to know if the item was 100% auto-matched by service criteria or manually matched by the user.  I think there is certainly room for improvement on both sides, the service, Wikidata, and any client like OpenRefine.   For services that take user edit uploads like Wikidata, it is very useful to know and even store with an edit upload "how" an item was reconciled with a service and other particulars, such as "what properties were used to provide disambiguation when the user reviewed and manually chose an item as **matched**".   We have already discussed multiple scenarios of even using machine learning for enhancements with a feedback loop with the user, but these are still early days.  Much of that discussion happens generally in our monthly meetups towards improving the [Reconciliation Service API](https://reconciliation-api.github.io/specs/latest/) which you might help with further ideas and join us.

Properties play a big part in disambiguation, and further still "the right properties for the right context or column type".  Here also we could make improvements on both sides, client and service, to ensure that the user is given good feedback to make wiser choices.  I would also strongly note:  Wikidata is a wiki and mentioned to me many times in the past... http://wiki.c2.com/?WikiDesignPrinciples= which is still the false utopia we live in and love for it's ideals.  But we certainly can do several things to make it a closer utopia for everyone wanting to keep facts accurate and timely.

**Comment by trnstlntk** [comment]
Suggestion to think more about, by Roderic Page (posted in the Wikiproject Biodiversity Telegram group and reposted here with his permission): 

> As an outsider who has used OpenRefine but not for Wikidata, perhaps it is an idea to work with a blacklist of types that can't be automatically merged (e.g., [Q16521 / taxon](https://www.wikidata.org/entity/Q16521)) plus rules requiring merges between items with the same values for P31 (taking into account that there might be a hierarchy of types such that the vales for P31 might not be identical as strings).

**Comment by wetneb** [comment]
Hi @andrawaag and @egonw,

I hear you, sloppy imports are definitely annoying. This problem you are reporting is at a fairly high level, so I can think of a lot of different approaches to tackle it: changing the Wikidata reconciliation service, changing OpenRefine's reconciliation workflows, changing the Wikidata extension, or even changing Wikidata's own data import policies and reviewing workflows.

Reconciling with names only can make sense in some cases, for instance when combined with a narrow enough type constraint, depending on the domain. On the other hand, reconciling with additional properties is not a guarantee to have more reliable matches either: it really depends on the case at hand. So I would be cautious about forbidding in the software itself the upload of reconciled values without additional properties as a blanket measure.

One other safety check that comes to mind here is to improve our quality assurance tools to detect the type mismatches that you report (gene vs cultural heritage object). Perhaps that could have caught the eye of the importer and prevented them from making this mistake?

I am also generally interested in improving the reconciliation workflows and making it easier for users to have greater assurances about the quality of their matches, but that is a big overarching goal that is broken down into a lot of small tasks, none of which is going to be a silver bullet to have perfect reconciliation results.

**Comment by ostephens** [comment]
>So I would be cautious about forbidding in the software itself the upload of reconciled values without additional properties as a blanket measure.

I tend to agree here @wetneb - I could see being this strict sometimes blocking useful work.

Is there anything we could do to make the default behaviour stricter (e.g. requiring additional properties as a default and then have an option to override to the current behaviour)? While this wouldn't guarantee anything, it would be a huge warning sign for the user they were doing something that could have some bad consequences - along the lines of the warning suggested by @egonw but with an additional step to override making the user think carefully about proceeding with the process?

**Comment by antoine2711** [comment]
@andrawaag: I know it not much, but at least, we should remove the default check on the feature.

Here's an issue about that, if you want to vote for it: #4722

Regards, Antoine
